vendor/github.com/longhorn/longhorn-spdk-engine/pkg/client/client.go

```diff
@@ -777,6 +902,31 @@ func (c *SPDKClient) EngineSnapshotHashStatus(name, snapshotName string) (respon
 	})
 }
 
+func (c *SPDKClient) EngineSnapshotClone(name, snapshotName, srcEngineName, srcEngineAddress string, cloneMode spdkrpc.CloneMode) error {
+	if err := util.VerifyParams(
+		util.Param{"name", name},
+		util.Param{"snapshotName", snapshotName},
+		util.Param{"srcEngineName", srcEngineName},
+		util.Param{"srcEngineAddress", srcEngineAddress},
+	); err != nil {
+		return errors.Wrap(err, "failed to clone snapshot for SPDK engine")
+	}
+
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
```

Comment:
Seems `grpcTimeoutSeconds` is logged in the caller but not enforced in the call. Should we apply this to the around the RPC call?

Reply:
Might be safer if we have a context timeout?

Reply:
Yes, we use the defaule timeout: 

```
ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
```